test/check-machines-lifecycle

```diff
@@ -254,73 +255,101 @@ class TestMachinesLifecycle(VirtualMachinesCase):
 
-        # newer libvirtd versions use socket activation
-        # we should test that separately, but here we test only using the service unit
-        if m.image not in ["debian-stable", "rhel-8-4", "rhel-8-5", "centos-8-stream"]:
-            m.execute("systemctl stop libvirtd-ro.socket libvirtd.socket libvirtd-admin.socket")
-            self.addCleanup(m.execute, "systemctl start libvirtd-ro.socket libvirtd.socket libvirtd-admin.socket")
+        def disableLibvirtService():
+            m.execute("systemctl disable {0}".format(libvirtServiceName))
 
         def hack_libvirtd_crash():
             # work around libvirtd crashing when stopped too quickly; https://bugzilla.redhat.com/show_bug.cgi?id=1828207
             m.execute("virsh domifaddr 1")
 
-        m.execute("systemctl disable {0}".format(libvirtServiceName))
+        def waitEmptyState():
+            with b.wait_timeout(15):
+                b.wait_in_text(".pf-c-empty-state", "Virtualization service (libvirt) is not active")
+
+        # Test that empty state appears when stopping libvirt service while page was loaded
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+
+        b.wait_in_text("body", "Virtual machines")
+        self.waitVmRow("subVmTest1")
+
+        disableLibvirtService()
         hack_libvirtd_crash()
-        m.execute("systemctl stop {0}".format(libvirtServiceName))
+        stopLibvirtSocket()
+        stopLibvirtService()
 
-        b.wait_in_text(".pf-c-empty-state", "Virtualization service (libvirt) is not active")
+        waitEmptyState()
+
+        # Test that starting libvirt service from the UI updates the page
         b.wait_visible("#enable-libvirt:checked")
         b.click(".pf-c-empty-state button.pf-m-primary")  # Start libvirt
         b.wait(lambda: checkLibvirtEnabled())
         b.wait_in_text("body", "Virtual machines")
-        with b.wait_timeout(15):
-            b.wait_not_present(".pf-c-empty-state")
-            self.waitVmRow("subVmTest1")
+        b.wait_not_present(".pf-c-empty-state")
+        self.waitVmRow("subVmTest1")
 
-            # Ensure that create VM dialog does not crash after starting libvirt from the UI
-            # https://bugzilla.redhat.com/show_bug.cgi?id=1974228
-            b.click("#create-new-vm")
-            b.wait_visible("#create-vm-dialog")
+        # Ensure that create VM dialog does not crash after starting libvirt from the UI
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1974228
+        b.click("#create-new-vm")
+        b.wait_visible("#create-vm-dialog")
 
-            self.browser.switch_to_top()
-            self.browser.wait_not_visible("#navbar-oops")
+        b.switch_to_top()
+        b.wait_not_visible("#navbar-oops")
 
-            self.browser.reload()
-            self.browser.enter_page('/machines')
-            self.browser.wait_in_text("body", "Virtual machines")
+        b.reload()
```

Comment:
why the reload? IMHO every reload in the tests should have a justifying comment at least.

Reply:
reload basically ensures that the app does not have a state already, ensuring that the next part of the test in (in matters of app state) independent from the previous parts.

It's especially important here, because we:
* have loaded the VMs list in the previous steps (subVmTest1 is present in the UI)
* I want to check in the next steps that starting from the empty state (stopped libvirtd) and then start the service from the UI will still fetch resources, thus make subVmTest1 appear in the UI. This was buggy before.

https://bugzilla.redhat.com/show_bug.cgi?id=1974223#c3

Will add the comment in the test as well.